compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -61,7 +61,9 @@ pub(crate) fn optimize_into_acir(
         .run_pass(Ssa::remove_bit_shifts, "After Removing Bit Shifts:")
         // Run mem2reg once more with the flattened CFG to catch any remaining loads/stores
         .run_pass(Ssa::mem2reg, "After Mem2Reg:")
+        .run_pass(Ssa::remove_enable_side_effects, "After EnableSideEffects removal:")
         .run_pass(Ssa::fold_constants, "After Constant Folding:")
+        .run_pass(Ssa::remove_enable_side_effects, "After EnableSideEffects removal:")
```

Comment:
Why do we run this pass both before and after constant folding?

Reply:
I think I placed the first one and then noticed that we needed some constant folding in order to catch everything so added another.

Seems like just the post-folding one is sufficient so I've removed the first.